ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/PaywallDialog.kt

```diff
@@ -84,8 +84,7 @@ private fun DialogScaffold(paywallOptions: PaywallOptions) {
 @Composable
 @ReadOnlyComposable
 private fun getDialogMaxHeightPercentage(): Float {
-    val aspectRatio = LocalConfiguration.current.screenHeightDp.toFloat() / LocalConfiguration.current.screenWidthDp
-    if (aspectRatio < UIDialogConstants.MAX_ASPECT_RATIO_TO_APPLY_MAX_HEIGHT) {
+    if (windowAspectRatio() < UIDialogConstants.MAX_ASPECT_RATIO_TO_APPLY_MAX_HEIGHT) {
```

Comment:
> I changed the algorithm to use the template size instead of the window size, so we don't care if it's displayed as a dialog, as full screen, or anything the developer wants to do. This will cause a recomposition unfortunately... But I think it's the best solution

Isn't this still the window size?

Reply:
Yeah, but in this case, we can indeed use the window size I think, since it's what's going to determine whether we use a full height dialog or not, so in this case, we do know it's going to be displayed as a dialog, and we already have some logic so we don't display it as a full screen dialog for MEDIUM and EXPANDED devices.

Also note that this is just a refactor, but the logic hasn't changed in this PR.

Reply:
Oooh right right 👍🏻